### PR TITLE
Several statsd logging bits

### PR DESCRIPTION
**Pull request by jerrymarino**
This PR contains several bits of statsd logging infrastructure.

In summary, it has the ability to log generation time and build time.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -74,8 +74,30 @@ buildifier(
     name = "buildifier"
 )
 
+script_base = "$SRCROOT/tools/instrumentation_helpers"
+
 xchammer_config = xchammer_config(
     targets=["//:xchammer"],
+    target_config={
```
Comment (jerrymarino) [inline BUILD.bazel]: This is how it logging is intended to be setup ( internally ), but probably will not end up being in XCHammer in this form

**Comment by mikezucc** [comment]
I like the usage of `xchammer_config` 👍

File: .gitignore [hunk 2]
```diff
@@ -37,3 +37,6 @@ bazel-*
 sample/*/*.xcodeproj
 external
 xchammer-bazel.xcodeproj
+
+# Statsd client
+tools/instrumentation_helpers/statsd_client
```
Comment (jerrymarino) [inline .gitignore]: It never actually calls this anymore - remove.

File: tools/instrumentation_helpers/statsd_post_build_action.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+CLIENT="$SRCROOT/tools/instrumentation_helpers/statsd_post_build_action.py"
+python $CLIENT 2>&1 | cat > /tmp/x
```
Comment (jerrymarino) [inline tools/instrumentation_helpers/statsd_post_build_action.sh]: Probably don't need to redirect to here.

File: tools/instrumentation_helpers/statsd_post_build_action.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+def write_build_metric():
+    start_time_f = os.path.join(
+        os.environ.get('TARGET_BUILD_DIR'),'xchammer.build_start')
+    start_time = os.path.getmtime(start_time_f)
+    delta = ((time.time()-start_time)*1000)
+    write_tsd("xchammer.build", delta)
+
+
+def write_last_generation_metric():
+    last_generation_log = "/private/var/tmp/xchammer.log"
```
Comment (jerrymarino) [inline tools/instrumentation_helpers/statsd_post_build_action.py]: If this log can be localized to the build directory, that would be ideal.

**Pull request merged**